pkg/provider/provider_helpers.go

```diff
@@ -187,3 +201,29 @@ func GetAccessTokenWithRefreshToken(
 	}
 	return result.AccessToken, nil
 }
+
+func envNameFieldDescription(description, envName string) string {
+	return fmt.Sprintf("%s Can also be sourced from the `%s` environment variable.", description, envName)
+}
+
+// TODO(SNOW-1752043 - next pr): move to common with resources
+func possibleValuesListed[T ~string | ~int](values []T) string {
+	valuesWrapped := make([]string, len(values))
+	for i, value := range values {
+		valuesWrapped[i] = fmt.Sprintf("`%v`", value)
+	}
+	return strings.Join(valuesWrapped, " | ")
+}
+
+func normalizeValidation[T any](normalize func(string) (T, error)) schema.SchemaValidateDiagFunc {
+	return func(val interface{}, _ cty.Path) diag.Diagnostics {
+		_, err := normalize(val.(string))
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		return nil
+	}
+}
+
+// validateBooleanString is similar to the one we use in resources, but in the provider config we also need to explicitly allow default value
+var validateBooleanString = resources.StringInSlice([]string{resources.BooleanTrue, resources.BooleanFalse, resources.BooleanDefault}, false)
```

Comment:
Let's distinguish these two validation not only in comment but also by name (either by validateBooleanStringInProvider/InResource or validateBooleanSTringWithDefault - other options will be welcome too )

Reply:
Done.